### PR TITLE
jwk updates

### PR DESCRIPTION
**Pull request by zjma**
- Pull some shared JWK logic out into a crate and make it a dependency of pepper service and jwk-consensus.
- Start fetching apple jwks in pepper service.

### CODE REVIEW

File: keyless/pepper/service/src/main.rs [hunk 1]
```diff
@@ -75,13 +75,8 @@ async fn main() {
...
     );
     jwk::start_jwk_refresh_loop(
-        "https://www.facebook.com",
-        "https://www.facebook.com/.well-known/oauth/openid/jwks",
-        Duration::from_secs(10),
-    );
-    jwk::start_jwk_refresh_loop(
-        "https://id.twitch.tv/oauth2",
-        "https://id.twitch.tv/oauth2/keys",
+        "https://appleid.apple.com",
```
Comment (danielxiangzl) [inline keyless/pepper/service/src/main.rs:78]: So these are not on-chain?

Comment (zjma) [reply keyless/pepper/service/src/main.rs:78]: not yet

**Review by danielxiangzl — APPROVED** [review]

File: crates/aptos-jwk-consensus/src/jwk_manager/mod.rs [hunk 2]
```diff
@@ -103,15 +103,27 @@ impl JWKManager {
...
+                match (maybe_issuer, maybe_config_url) {
+                    (Ok(issuer), Ok(config_url)) => Some(JWKObserver::spawn(
+                        self.epoch_state.epoch,
+                        self.my_addr,
+                        issuer,
+                        config_url,
+                        Duration::from_secs(10),
+                        local_observation_tx.clone(),
+                    )),
+                    (maybe_issuer, maybe_config_url) => {
```
Comment (alinush) [inline crates/aptos-jwk-consensus/src/jwk_manager/mod.rs:119]: I guess this syntax helps you avoid specifying all combinations of `None/Some`?

Comment (zjma) [reply crates/aptos-jwk-consensus/src/jwk_manager/mod.rs:119]: exactly!

File: crates/aptos-jwk-consensus/src/jwk_manager/mod.rs [hunk 3]
```diff
@@ -103,15 +103,27 @@ impl JWKManager {
...
-            .map(|provider| {
-                JWKObserver::spawn(
-                    self.epoch_state.epoch,
-                    self.my_addr,
-                    provider.name.clone(),
-                    provider.config_url.clone(),
-                    Duration::from_secs(10),
-                    local_observation_tx.clone(),
-                )
+            .flat_map(|provider| {
```
Comment (alinush) [inline crates/aptos-jwk-consensus/src/jwk_manager/mod.rs:106]: Why the `flat_map`? You are mapping `provider` to one value, not to an iterator over values no?

File: crates/aptos-jwk-consensus/src/jwk_observer.rs [hunk 4]
```diff
@@ -90,29 +51,33 @@ impl JWKObserver {
...
+        open_id_config_url: String,
         observation_tx: aptos_channel::Sender<(), (Issuer, Vec<JWK>)>,
         close_rx: oneshot::Receiver<()>,
     ) {
-        let issuer_str =
-            String::from_utf8(issuer.clone()).unwrap_or_else(|_e| "UNKNOWN_ISSUER".to_string());
         let mut interval = tokio::time::interval(fetch_interval);
         interval.set_missed_tick_behavior(MissedTickBehavior::Delay);
         let mut close_rx = close_rx.into_stream();
+        let my_addr = if cfg!(feature = "smoke-test") {
```
Comment (alinush) [inline crates/aptos-jwk-consensus/src/jwk_observer.rs:62]: What is this for? There are no comments on this function.

**Review by alinush — APPROVED** [review]

File: crates/aptos-jwk-consensus/src/jwk_manager/mod.rs (hunk 3 above)
Comment (zjma) [reply crates/aptos-jwk-consensus/src/jwk_manager/mod.rs:106]: looks like i meant `filter_map` 🤦

**Pull request merged**